ts_src/psbt.ts

```diff
@@ -733,13 +739,12 @@ export class Psbt {
     });
 
     // generate output blinding factors
-    const numOutputs = this.data.outputs.length - 1; // exclude fees output
-    const outputAbfs = this.data.outputs
-      .slice(0, numOutputs)
-      .map(() => randomBytes(opts!));
-    const outputVbfs = this.data.outputs
-      .slice(0, numOutputs - 1)
-      .map(() => randomBytes(opts!));
+    const numOutputs = c.__TX.outs.length - 1 // excluding fee output
+    const outputAbfs = range(numOutputs)
+      .map((index: number) => outputsIndexToBlind!.includes(index) ? randomBytes(opts!) : Buffer.alloc(32));
```

Comment:
at this point, we need to generate the random blinders (so-called blinding factors) for both asset and value of each output that is gonna be blinded. Hence, we could just create outputAbfs (and also outputVbfs) like:
```suggestion
    const outputAbfs = outputsIndexToBlind!.map(() => randomBytes(opts!));
```